File: 실버 5 (4문제)/4108_지뢰 찾기.c
// 참고 링크 : https://moon-gd.tistory.com/38
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#define MAX 101

int dirRow[8] = { -1, -1, -1, 0, 1, 1, 1, 0 };
int dirCol[8] = { -1, 0, 1, 1, 1, 0, -1, -1 };
char road[MAX][MAX];
int row, col;

int main() {

	while (1) {
		// 행과 열 입력받기
		scanf("%d %d", &row, &col);

		// 둘 다 0이 입력되면 종료
		if (row + col == 0) {
			break;
		}
		
		for (int i = 0; i < row; i++) {
			scanf("%s", road[i]);
		}

		for (int i = 0; i < row; i++) {
			for (int j = 0; j < col; j++) {
				// 지뢰인 경우 continue
				if (road[i][j] == '*') {
					continue;
				}
				// 지뢰가 아닌 경우, 주변 지뢰의 개수 확인
				int count = 0;
				for (int k = 0; k < 8; k++) {
					// 유효 범위를 벗어난 경우 continue
					if (dirRow[k] + i < 0 || dirRow[k] + i >= row || dirCol[k] + j < 0 || dirCol[k] + j >= col) {
						continue;
					}

					// 유효 범위 내에서 지뢰가 있을 경우 count 1 증가
					if (road[i + dirRow[k]][j + dirCol[k]] == '*') {
						count++;
					}
				}
				road[i][j] = '0' + count;
			}
		}

		for (int i = 0; i < row; i++) {
			printf("%s\n", road[i]);
		}
	}


	return 0;
}

File: 실버 5 (7문제)/4108_지뢰 찾기.py
# 참고 링크 : https://moon-gd.tistory.com/38
dirRow = [-1, -1, -1, 0, 1, 1, 1, 0]
dirCol = [-1, 0, 1, 1, 1, 0, -1, -1]

while True:
    row, col = map(int, input().split())

    # 종료 조건 (둘 다 0인 경우)
    if row + col == 0:
        break

    road = []
    # 입력 받기
    for _ in range(row):
        road.append(input())

    for i in range(row):
        for j in range(col):
            if road[i][j] == '*':
                road[i] += '*'
                continue
            else:
                count = 0
                for k in range(8):
                    # 허용 범위의 행에 존재할 때
                    if i + dirRow[k] < 0 or i + dirRow[k] >= row:
                        continue
                    # 허용 범위의 열에 존재할 때
                    if j + dirCol[k] >= col or j + dirCol[k] < 0:
                        continue
                    # 허용 범위 안에서 지뢰가 존재할 때
                    if road[i + dirRow[k]][j + dirCol[k]] == '*':
                        count += 1
            road[i] += str(count)

    for i in range(row):
        print(road[i][col:])
